app/src/main/java/dev/andrylat/app/App.java

```diff
@@ -6,21 +6,62 @@
 import java.util.stream.IntStream;
 
 public class App {
-    private static StringBuilder errors = new StringBuilder();
+    private StringBuilder errors;
+    private Scanner scanner;
+    private String brand = "";
+
+    public App() {
+        System.out.println("Hello. Enter card number for validation:");
+        errors = new StringBuilder();
+        scanner = new Scanner(System.in);
+    }
 
     public static void main(String[] args) {
-        var scanner = new Scanner(System.in);
+        new App().process().print();
+    }
 
-        System.out.println("Hello. Enter card number for validation:");
+    public App process() {
+        var input = scanner.nextLine();
+        
+        validate(input);
+
+        var cardNum = getCardNum(input);
+        determineBrandByNumber(cardNum);
+        validate(cardNum);
+
+        return this;
+    }
+
+    private void determineBrandByNumber(ArrayList<Integer> cardNum) {
+        /*
+           for later improvements:
+
+           I could create an enum with all card brands,
+           and each card brand would have a method to validate
+           if the the given number equals to the brand's possible number
+           the return type of the method would be boolean
 
-        var cardNum = new ArrayList<>(Arrays.stream(scanner.nextLine().split(""))
-                         .filter(App::isDigit)
+        */
+
+        brand = cardNum.get(0) == 4 ? "VISA" : cardNum.get(0) == 5 && IntStream.range(0,6).anyMatch(n -> n == cardNum.get(1)) ? "MASTERCARD" : "";
+    }
+
+    private ArrayList<Integer> getCardNum(String s) {
+        return new ArrayList<>(Arrays.stream(s.split(""))
+                         .filter(this::isDigit)
                          .map(Integer::valueOf)
                          .toList());
+    }
+
+    private void validate(String input) {
+        if (input.length() < 16) {
```

Comment:
What is 16? Why exactly 16? I don't understand.

For clear explanation from your code you should create named constant with proper explainable name.